.github/workflows/build.yml

```
@@ -0,0 +1,27 @@
+name: Build
+
+on:
+  pull_request:
+    types: [ labeled ]
```

If I'm not mistaken, this will only trigger this workflow **once** when the PR is labeled (immediately after it got created). It won't trigger the workflow for future events like rebases, etc. I suggest we add other event types and make the label check more robust. See this SO answer: https://stackoverflow.com/questions/62325286/run-github-actions-when-pull-requests-have-a-specific-label#comment122159108_62331521

---

Good hint, will add it 👍

---

So it's  `[opened, synchronize, labeled, reopened]` I guess?

```suggestion
    types: [ opened, synchronize, labeled, reopened ]
```

---

According to the SO answer these seem to be the most important ones, yes. But with these new event types we need to also change the `if` logic in the `build` job because most of these events won't have the `event.label.name` property so the workflow will throw an error. We need to use the `if: contains(github.event.pull_request.labels.*.name, 'dependabot')` logic.